vsphere/data_source_vsphere_virtual_machine.go

```diff
@@ -109,6 +109,10 @@ func dataSourceVSphereVirtualMachineRead(d *schema.ResourceData, meta interface{
 		return fmt.Errorf("no configuration returned for virtual machine %q", vm.InventoryPath)
 	}
 
+	if props.Config.Uuid == "" {
+		return fmt.Errorf("configuration returned for virtual machine %q is scrappy: any UUID", vm.InventoryPath)
```

Comment:
Can this error message be changed?  I'd suggest something like:

```
return fmt.Errorf("virtual machine %q does not have a UUID", vm.InventoryPath)
```

Reply:
Agree with you, let's go.